multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/SystemDomain.java

```diff
@@ -34,13 +35,15 @@
   public static final String SYSTEM_DOMAIN_ACTIONS = "system.domain.actions";
   public static final String DOMAIN_ID = "system";
   public static final String MANAGE_SYSTEM_ACTION = "manageSystem";
+  public static final String MONITOR_SYSTEM_ACTION = "monitorSystem";
 
   @Inject
   public SystemDomain(@Named(SYSTEM_DOMAIN_ACTIONS) Set<String> allowedActions) {
     super(
         DOMAIN_ID,
-        Stream.concat(allowedActions.stream(), Stream.of(MANAGE_SYSTEM_ACTION))
-            .collect(Collectors.toList()),
+        Stream.concat(
+                allowedActions.stream(), Stream.of(MANAGE_SYSTEM_ACTION, MONITOR_SYSTEM_ACTION))
```

Comment:
Additional system action may be configured instead of hardcoded, how it is done with [manageUsers](https://github.com/eclipse/che/blob/3a14bacda15ad501b77161abc7c70dec5b917913/assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java#L371-L373). But I agree that `monitorSystem` action is a common one and it's OK to hardcode it. Just note that to make sure that you are aware of binding system actions.

Reply:
I opted for hardcoding because I saw this as common for both single and multiuser environments. I thought that the dynamic binding is used more for the case where the set of actions differs between single and multiuser. E.g. your example doesn't really make sense in the single-user deployment.

Reply:
Single-user is not a case at all. There is no any authentication nor authorization. And as far as you can see, `PermissionsModule` is bound only for multiuser https://github.com/eclipse/che/blob/3a14bacda15ad501b77161abc7c70dec5b917913/assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java#L358

And why we have hardcoded and dynamic bindings: In all permissions domains except system we use only hardcoded actions list because each domain knows its list of all known actions.

And there was an idea that `system domain` protects admin users methods, and it is not able to depend on all modules (because of circular dependencies, conceptual issues). And then each module can provide its own action to protect its Service methods. Now it is used by `user-api` module to provide an ability to `manageUsers`. As far as I remember previously we used to use `managerOrganization` action to make organizations related methods accessible for admins (but not all, only that one who has this particular action).